homeassistant/components/glinet/__init__.py

```diff
@@ -0,0 +1,73 @@
+"""The GL-inet integration."""
+from __future__ import annotations
+
+from homeassistant.components.device_tracker.const import CONF_CONSIDER_HOME
+from homeassistant.config_entries import SOURCE_IMPORT, ConfigEntry
+from homeassistant.core import HomeAssistant
+
+from .const import DATA_GLINET, DOMAIN
+from .router import GLinetRouter
+
+PLATFORMS = ["device_tracker"]
+
+
+async def async_setup(hass, config):
+    """Set up the GLinet integration."""
+    conf = config.get(DOMAIN)
+    if conf is None:
+        return True
+
+    # save the options from config yaml
+    options = {}
+    for name, value in conf.items():
+        if name in ([CONF_CONSIDER_HOME]):
+            options[name] = value
+    hass.data[DOMAIN]["yaml_options"] = options
+
+    # check if already configured
+    domains_list = hass.config_entries.async_domains()
+    if DOMAIN in domains_list:
+        return True
+
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_IMPORT}, data=conf
```

Comment:
We only support import for integrations that migrate to config flow. For new ones drop `async_setup`.